### PR TITLE
Add settings for alertmanager clusters

### LINKED ISSUE #38: Add PodMonitor
We should have a PodMonitor to, using the `pod` role in the Prometheus config. That solves monitoring pods running the same application but which are only partially belong to a service, e.g. quarantined pods.
This should be easy enough and would look exactly like the service monitor only with different selector semantics:

```yaml
apiVersion: monitoring.coreos.com/v1alpha1
kind: PodMonitor
metadata:
  name: frontend
  labels:
    tier: frontend
spec:
  selector:
    # label selector as in ServiceMonitor but doesn't select services of which we discover pods
    # but directly selects pods.
  endpoints:
    # similar to ServiceMonitor, just we don't differentiate between target ports and service ports.
```

This now has to be included into a Prometheus TPR and would happen just like for ServiceMonitors (x-ref: #37).
Now we have two selectors in there (or two lists of selectors) to select different resource types. Another option would be to have a single selector applying to ServiceMonitors and PodMonitors. A single selector applying to two different resource types is quite implicit and hence not too great. OTOH, as a user I don't really care about what type the monitor is.

Another option would be to consolidate both in a single Monitor.

```yaml
apiVersion: monitoring.coreos.com/v1alpha1
kind: Monitor
metadata:
  name: frontend
  labels:
    tier: frontend
spec:
  pods:
    # label selector working on pods
  services:
    # label selector working on service endpoints
  endpoints:
    # similar to ServiceMonitor, just we don't differentiate between target ports and service ports.
```

Seems interesting first but I believe it will constrain us in the long run by colliding options that only apply to either, i.e. I'd like to allow specifying service blackbox probing at some point.

Thoughts on any of these options and ideas?

@matthiasr @grobie @brian-brazil @brancz

### PR DESCRIPTION
**Pull request by hacktastic**
alertmanager: adds settings for alertmanager statefulsets

This PR exposes three new alertmanager flags:
- --cluster.gossip-interval
- --cluster.pushpull-interval
- --cluster.peer-timeout

Each flag has a corresponding field on AlertmanagerSpec:
- clusterGossipInterval
- clusterPushpullInterval
- clusterPeerTimeout

### CODE REVIEW

File: pkg/apis/monitoring/v1/types.go [hunk 1]
```diff
@@ -1283,6 +1283,12 @@ type AlertmanagerSpec struct {
 	// Needs to be provided for non RFC1918 [1] (public) addresses.
 	// [1] RFC1918: https://tools.ietf.org/html/rfc1918
 	ClusterAdvertiseAddress string `json:"clusterAdvertiseAddress,omitempty"`
+	// Interval between gossip attempts
+	ClusterGossipInterval string `json:"clusterGossipInterval,omitempty"`
+	// Interval between pushpull attempts
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1/types.go:1288]:
```suggestion
	// Interval between pushpull attempts.
```

File: pkg/apis/monitoring/v1/types.go [hunk 2]
```diff
@@ -1283,6 +1283,12 @@ type AlertmanagerSpec struct {
 	// Needs to be provided for non RFC1918 [1] (public) addresses.
 	// [1] RFC1918: https://tools.ietf.org/html/rfc1918
 	ClusterAdvertiseAddress string `json:"clusterAdvertiseAddress,omitempty"`
+	// Interval between gossip attempts
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1/types.go:1286]:
```suggestion
	// Interval between gossip attempts.
```

File: pkg/apis/monitoring/v1/types.go [hunk 3]
```diff
@@ -1283,6 +1283,12 @@ type AlertmanagerSpec struct {
...
+	ClusterGossipInterval string `json:"clusterGossipInterval,omitempty"`
+	// Interval between pushpull attempts
+	ClusterPushpullInterval string `json:"clusterPushpullInterval,omitempty"`
+	// Timeout for cluster peering
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1/types.go:1290]:
```suggestion
	// Timeout for cluster peering.
```

**Review by simonpasquier — COMMENTED** [review]
Thanks! The linked issue doesn't look correct :)

**Comment by hacktastic** [comment]
@simonpasquier would you like me to squash and re-push?

**Review by simonpasquier — APPROVED** [review]
lgtm

**Pull request merged**